### ISSUE TITLE
Pyscf2qwalk with MCSCF WF

### ISSUE THREAD

**Issue by mcbennet** [lkwagner/autogenv2#28]
Hi, 

I have recently had difficulties reproducing a CASSCF result of PYSCF with a no-jastrow VMC calculation with QWalk -- let me mention right away that this appears to not be a coefficient tolerance issue as I see the discrepancy for arbitrarily small tolerances. 

The test system I am looking at is the benzene molecule. I am looking at 3 states --- the ground state (symmetry=Ag) and the singlets of B2u and B3u symmetry (calculated in D2h). The excitations that I obtain from PYSCF look very reasonable and agree with previous calculations in the literature.

If I then take the three pyscf2qwalk-converted wave functions and perform a no-jastrow vmc on each, individually, I see something bizarre.

STATE          PYSCF/CASSCF    QWALK/NOJ-VMC
       Ag    -36.6391780013251              -36.6415(9)
     B3u    -36.4548250877713              -36.4548(9)
     B2u    -36.3473573036637              -36.3742(9) 

Notice that 2/3 basically agree, while in the case of the B2u state, the disagreement is significant. It is also strange to me that the energy from QWalk is _lower_.  

I have mostly convinced myself that this is not a problem with the orbitals because I see the same thing when I run a CASCI calculation in PYSCF and feed the HF orbitals directly into the converter. If the maintainers have any suspicions for where the discrepancy of the B2u state is coming from and potentially even solutions for how I might achieve better agreement, I would be very grateful

My pyscf code, after defining the geometry and the basis, is the following:

mol.build()
mf = scf.RHF(mol)
mf.kernel()

cas_list = [11,14,15,16,17,24]
mycas = mcscf.CASSCF(mf, 6, 6)
mycas.fcisolver = fci.solver(mol,singlet=True)
cas_orbs = mycas.sort_mo(cas_list) # Sorted such that only the pi-orbs are active

\# For the different states, I simply change the wfnsym variable here (Ag,B3u,B2u)
mycas.fcisolver.wfnsym='B2u'

mycas.verbose = 4
mycas.run(cas_orbs)

qw.print_qwalk(mol,mycas,method='mcscf',tol=0.00001,basename='qwalk')

**Comment by lkwagner** [comment]
Hi Chandler,

Can you attach the full script so that we can check it? 

Thanks!

**Comment by mcbennet** [comment]
Certainly. Here is the full script 
[casscf.txt](https://github.com/lkwagner/autogenv2/files/2649785/casscf.txt)

**Comment by lkwagner** [comment]
I did reproduce this problem. Not sure what the issue is yet, though!

**Comment by lkwagner** [comment]
While I haven't resolved it, I suspect the issue is with the orbital reordering. Can you try just running with CASCI and not reordering the orbitals and see if that reproduces the problem?

**Comment by mcbennet** [comment]
The state that disagrees the most is the 2nd singlet excitation. I have looked at turning off D2h symmetry and targeting the state of interest in the following way:

cas_list = [11,14,15,16,17,24]
mycas = mcscf.CASSCF(mf, 6, 6)
mycas.fcisolver = fci.solver(mol,singlet=True)
mycas.fcisolver.nroots=3
mycas.state_specific_(2)
cas_orbs = mycas.sort_mo(cas_list)

The energy that comes out of pyscf is identical to my energy before, however, I find that a no-jastrow VMC with the resulting wf agrees much more closely. 

Total Energy:  -36.34725695 +/-  0.004508095993 
(high error here but clearly not the same state from before)

In some way which I don't quite yet understand, I suppose pyscf (in my original script) was handing qwalk a WF that mixed in a lower state possibly.

**Comment by lkwagner** [comment]
I don't think that pyscf2qwalk understands symmetry in the CI expansion, so that very well may be it.

**Comment by mcbennet** [comment]
Also, I have plotted the orbitals in qwalk and they are the correct ones, namely, the pi orbitals which I was intending to excite into.

**Comment by lkwagner** [comment]
If you happen to know how to do it, we'd very much like to know; the function in pyscf2qwalk is print_cas_slater. :)
